docs/csharp/language-reference/builtin-types/value-tuples.md

```diff
@@ -66,6 +63,14 @@ At compile time, the compiler replaces non-default field names with the correspo
 > [!TIP]
 > Enable .NET code style rule [IDE0037](../../../fundamentals/code-analysis/style-rules/ide0037.md) to set a preference on inferred or explicit tuple field names.
 
+Beginning with C# 12, you can specify an alias for a tuple type with a [`using` statement](../keywords/using-directive.md#using-alias). The following example adds a `global using` alias for a Tuple type with two integer values for an allowed `Min` and `Max` value:
+
+:::code language="csharp" source="snippets/shared/ValueTuples.cs" id="AliasTupleType":::
+
+After declaring the alias, you can use the `BandPass` name as an alias for that tuple type:
```

Comment:
I have two questions related to aliasing tuple types.

Question 1. Does alias hold information about tuple field names? That is, if I have two aliases like following:
```csharp
global using BandPass = (int Min, int Max);
global using KeyValue = (int Key, int Value);
```

and there is a method that accepts `BandPass` parameter, can I pass a `KeyValue` argument to this method? If yes, then the code looks type safe, while it's not. (of course it's type safe, because actual types here are `ValueTuple<int, int>`)

Question 2. Why should I define a (global) alias for a tuple type and not a positional `record struct` instead:
```csharp
public record struct BandPass(int Min, int Max);
```

Semantics looks the same (equality by component values, `record struct` instances are mutable, deconstruction), but additionally we get type safety. Possible performance overhead may be ignored in the majority of cases, I believe. Have I missed anything here? Would it be fine to say: "If you want to define an alias for a tuple type, consider defining a `record struct` instead."?

Should the docs elaborate on these two questions and give any advice (especially for the second question)?

Reply:
Addressed in the most recent commit.

Thanks @pkulikov Those are great questions that needed clarification.